packages/editor-preview/src/browser/editor-preview-tree-decorator.ts

```diff
@@ -49,6 +56,35 @@ export class EditorPreviewTreeDecorator implements TreeDecorator, FrontendApplic
         this.shell.onDidAddWidget(widget => this.registerEditorListeners(widget));
         this.shell.onDidRemoveWidget(widget => this.unregisterEditorListeners(widget));
         this.editorWidgets.forEach(widget => this.registerEditorListeners(widget));
+        this.fileService.onDidFilesChange(async (event: FileChangesEvent) => {
+            if (!this.corePreferences['workbench.editor.closeOnFileDelete']) {
+                await this.checkForEditorDeleted(event);
+                this.fireDidChangeDecorations((tree: Tree) => this.collectDecorators(tree));
+            }
+        });
+    }
+
+    protected async checkForEditorDeleted(event: FileChangesEvent): Promise<void[] | undefined> {
```

Comment:
Perhaps `Promise<void>` and just `await Promise.all(fileExistPromises)` as the last line?

Reply:
👍